doc/src/tutorials/physics/mechanics/duffing-example.rst

```diff
@@ -0,0 +1,141 @@
+.. figure:: duffing-oscillator-pendulum.svg
+
+The system will be modeled using Lagrange equations. `M` is mass of the Duffing oscillator,
+`m` is mass of the pendulum, `l` is length of the pendulum. `k_1` and `k_2` are linear and
+non-linear parts of spring stiffness, and `c_1` is a viscous damping coefficient of the Duffing oscillator.
+
+   >>> import sympy as sm
+   >>> import sympy.physics.mechanics as me
+
+Define Variables
+================
+
+   >>> M, m, l, k1, k2, c1, g, h, w, d, r = sm.symbols('M, m, l, k1, k2, c1, g, h, w, d, r')
+   >>> q1, q2 = me.dynamicsymbols('q1 q2')
+   >>> q1d, q2d = me.dynamicsymbols('q1 q2', 1)
+   >>> u1, u2 = me.dynamicsymbols('u1 u2')
+   >>> u1d, u2d = me.dynamicsymbols('u1 u2', 1)
+
+- :math:`h`: Height of the Duffing oscillator
+- :math:`w`: Width of the Duffing oscillator
+- :math:`d`: Depth of the Duffing oscillator
+- :math:`r`: Radius of the massive bob of the pendulum
+- :math:`q_1`: Generalized coordinate representing the position of the Duffing oscillator
+- :math:`q_2`: Generalized coordinate representing the angle of the pendulum
+- :math:`\dot{q}_1`: First time derivative of `q1`, representing the velocity of the Duffing oscillator
+- :math:`\dot{q}_2`: First time derivative of `q2`, representing the angular velocity of the pendulum
+- :math:`u_1`: Generalized speed associated with the Duffing oscillator
+- :math:`u_2`: Generalized speed associated with the pendulum
+- :math:`\dot{u}_1`: First time derivative of `u1`, representing the acceleration of the Duffing oscillator
+- :math:`\dot{u}_2`: First time derivative of `u2`, representing the angular acceleration of the pendulum
+
+Define Kinematics
+=================
+
+Define all the reference frames and points.
+
+   >>> N = me.ReferenceFrame('N')
+   >>> B = N.orientnew('B', 'axis', (q2, N.z))
+
+The angular velocity of the pendulum in the reference frame is:
+
+   >>> B.set_ang_vel(N, q2.diff() * N.z)
+   >>> B.ang_vel_in(N)
+   Derivative(q2(t), t)*N.z
+
+Locations and velocities of the Duffing Oscillator block and the pendulum are:
+
+   >>> O = me.Point('O')
+   >>> Block = O.locatenew('Block', q1 * N.y)
+   >>> Pendulum = Block.locatenew('Pendulum', l * B.y)
+
+O is a fixed point in the inertial reference frame.
+
+   >>> O.set_vel(N, 0)
+   >>> Block.set_vel(N, q1d * N.y)
+   >>> Pendulum.v2pt_theory(Block, N, B)
+   Derivative(q1(t), t)*N.y - l*Derivative(q2(t), t)*B.x
+
+   >>> me.init_vprinting()
+
+Define inertia and rigid bodies.
+Here, we assume a simple pendulum which consists of a bob of mass m hanging from a massless string of length l
+and fixed at a pivot point (Duffing Oscillator Block).
+
+   >>> I_block = me.inertia(N, M*(h**2 + d**2)/12, M*(w**2 + h**2)/12, M*(w**2 + d**2)/12)
+   >>> I_pendulum = me.inertia(B, 0, 0, 2*m*r**2/5)
+
+   >>> par_block = me.RigidBody('block', Block, N, M, (I_block, Block))
+   >>> par_pendulum = me.RigidBody('pendulum', Pendulum, B, m, (I_pendulum, Pendulum))
```

Comment:
I would propose just using particles as done in the paper. This allows comparing the final equations to the paper more easily and it makes the equations more readable as they are simpler.

When doing so, don't forget to remove `h`, `d`, `w`, and `r` from the start of the example.

Reply:
I compared the equations of motion derived from `LagrangesMethod `.
Equations of motion in the paper:
$$(M+m)q_{1}'' - mlq_{2}''sin(q_{2})-ml(q_{2}')^2cos(q_{2})+k_{1}q_{1}+k_{2}(q_{1})^3+c_{1}q_{1}'$$
$$ml^2q_{2}''-mlq_{1}''sin(q_{2})+mlgsin(q_{2})+c_{2}q_{2}'$$
Equations of motion in this example:
$$(M+m)q_{1}'' - mlq_{2}''sin(q_{2})-ml(q_{2}')^2cos(q_{2})+k_{1}q_{1}+k_{2}(q_{1})^3+c_{1}q_{1}' -(M+m)g$$
$$ml^2q_{2}''-mlq_{1}''sin(q_{2})+mlgsin(q_{2})+ \frac{2r^2q_{2}''}{5}$$

The differences come from the gravitational force and the existence of the damping torque with damping coefficient $c_2$, so I believe equations of motion obtained in this example are acceptable

Reply:
Part of the reason was that it also kept the tutorial simpler (the block doesn't rotate so those inertia terms are by definition not showing up in any of the equations), but keeping it is fine with me.